thehive/app/org/thp/thehive/controllers/v1/PatternCtrl.scala

```diff
@@ -0,0 +1,127 @@
+      patternSrv
+        .startTraversal(graph)
+  )
+  override val pageQuery: ParamQuery[OutputParam] = Query.withParam[OutputParam, Traversal.V[Pattern], IteratorOutput](
+    "page",
+    FieldsParser[OutputParam],
+    (range, patternSteps, _) => patternSteps.richPage(range.from, range.to, range.extraData.contains("total"))(_.richPattern)
+  )
+  override val outputQuery: Query = Query.output[RichPattern, Traversal.V[Pattern]](_.richPattern)
+  override val getQuery: ParamQuery[EntityIdOrName] = Query.initWithParam[EntityIdOrName, Traversal.V[Pattern]](
+    "getPattern",
+    FieldsParser[EntityIdOrName],
+    (idOrName, graph, _) => patternSrv.get(idOrName)(graph)
+  )
+
+  def importMitre: Action[AnyContent] =
+    entrypoint("import MITRE ATT&CK patterns")
+      .extract("file", FieldsParser.file.on("file"))
+      .authPermitted(Permissions.managePattern) { implicit request =>
+        val file: FFile = request.body("file")
+
+        for {
+          inputPatterns <- parseJsonFile(file)
+          richPatterns =
+            inputPatterns
+              .sortBy(_.external_id.length) // sort to create sub-patterns after their parent
+              .foldLeft[JsArray](JsArray.empty) { (array, inputPattern) =>
+                val res = db.tryTransaction { implicit graph =>
+                  createFromInput(inputPattern)
+                } match {
+                  case Failure(e) =>
+                    Json.obj("status" -> "Failure", "message" -> e.getMessage)
+                  case Success(t) =>
+                    Json.obj("status" -> "Success", "mitreId" -> t.patternId, "patternName" -> t.name)
+                }
+                array :+ res
+              }
+        } yield Results.Created(richPatterns)
+      }
+
+  def get(patternId: String): Action[AnyContent] =
+    entrypoint("get pattern")
+      .authRoTransaction(db) { implicit request => implicit graph =>
+        patternSrv
+          .get(EntityIdOrName(patternId))
+          .richPattern
+          .getOrFail("Pattern")
+          .map(richPattern => Results.Ok(richPattern.toJson))
+      }
+
+  def getCasePatterns(caseId: String): Action[AnyContent] =
+    entrypoint("get case patterns")
+      .authRoTransaction(db) { implicit request => implicit graph =>
+        for {
+          patternIds <- patternSrv.getCasePatterns(caseId)
+        } yield Results.Ok(patternIds.toJson)
+      }
+
+  def delete(patternId: String): Action[AnyContent] =
+    entrypoint("delete pattern")
+      .authPermittedTransaction(db, Permissions.managePattern) { implicit request => implicit graph =>
+        patternSrv
+          .getOrFail(EntityIdOrName(patternId))
+          .flatMap(patternSrv.remove)
+          .map(_ => Results.NoContent)
+      }
+
+  private def parseJsonFile(file: FFile): Try[Seq[InputPattern]] =
+    for {
+      stream <- Try(new FileInputStream(file.filepath.toString))
+      json = Json.parse(stream)
+    } yield (json \ "objects").get.as[Seq[InputPattern]]
```

Comment:
Why you `get` the `JsLookupResult` ?
```suggestion
    } yield (json \ "objects").as[Seq[InputPattern]]
```